cores/esp32/esp32-hal-uart.c

```diff
@@ -45,171 +37,67 @@
 #else 
 #error Target CONFIG_IDF_TARGET is not supported
 #endif
-#else // ESP32 Before IDF 4.0
-#include "rom/ets_sys.h"
-#include "rom/uart.h"
-#include "esp_intr.h"
-#endif
 
 #if CONFIG_IDF_TARGET_ESP32
 #define UART_PORTS_NUM 3
```

Comment:
in `soc/soc_caps.h` you can find defined `SOC_UART_NUM`. Just use that :)

Reply:
Good tip! Thank you.

Reply:
Done. Commited.